clusterloader2/api/default.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"fmt"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+// SetDefaults set the default configuration parameters.
+func (conf *Config) SetDefaults() {
+	//To support deprecated "automanangedNamespaces" in testconfig
+	conf.AutomanagedNamespaces = 1
+
+	conf.Namespace.SetDefaults()
+}
+
+// SetDefaults specifies the default values for namespace parameters.
+func (ns *NamespaceConfig) SetDefaults() {
+	ns.Number = 1
+	ns.Prefix = fmt.Sprintf("test-%s", util.RandomDNS1123String(6))
+	ns.DeleteStaleNamespaces = false
```

Comment:
Same here. And this has to be a pointer, because otherwise you have no way to distinguish between unset value and explicitly set to false.
[It's also a reason  why I previously commented about removing a pointer only for Number field.]

Reply:
Same here. The defaults are set using the command line flags or default values before parsing the yaml file and will be overridden by the values in testconfig file. This is an simple and clean implementation in my opinion.

Reply:
made the change